src/main/java/frontend/MessageConsole.java

```diff
@@ -0,0 +1,37 @@
+package frontend;
+
+import javax.swing.*;
+import javax.swing.text.Document;
+import java.awt.*;
+import java.io.PrintStream;
+
+
+//TODO: what does this class do?
```

Comment:
it realistically wouldn't be to difficult to extend the logger and have the log4j config direct a log stream at the custom extension.